### ISSUE TITLE
Passing empty AR Relation to policy_scope...

### ISSUE THREAD

**Issue by erlandsona** [elabs/pundit#328]
Hello I'm trying to use policy_scope to restrict access to the list of resources gotten from an Active Record Relation...

Here's what I did...
shortened code for app/controllers/api/v1/models/brands_controller.rb
```
def index
  @model = Model.find params[:model_id]
  authorize @model, :show?
  @brands = policy_scope(@model.brands)
  render json: @brands
end
```
shortened app/policies/application_policy.rb
```
class ApplicationPolicy
  ...
  class Scope
    attr_reader :client, :scope
    def initialize(client, scope)
      @client = client
      @scope = scope
    end
    def resolve
      if @client.admin?
        @scope.all
      else
        # return all resources with pending resources scoped to client.
        @scope.where("approved = true OR client_id = ?", @client.id)
      end
    end
  end
end
```
as you can see I have a typical nested resource with the controller name-spaced to give context to a HABTM relationship I have setup in the model layer (please forgive the unconventional naming I'm using in code... It's relevant for the specific domain).

So a client application will hit /brands/1/models or in this case /models/1/brands and the expected outcome is the list of brands that have the model with id=1... and visa versa...

There's two issues happening with this approach... https://gist.github.com/kidpollo/5287007

One is that it throws an error when the ARRelation is empty because the where clause can't find the approved column on an empty AR Relation...

Two is that if I add a condition to check for an empty AR Relation and I pass in a class name to the policy_scoped method then it wines because the class doesn't have the empty? method defined...

What is you're suggestion for scoping these nested associations like this?

**Comment by jnicklas** [comment]
Honestly, I have no idea what you're talking about. The solution for the linked gist is documented in the [README](https://github.com/elabs/pundit#additional-context).

But that doesn't seem like what you're wanting to do. I don't understand why you need to pass an empty scope to the Scope class, but if you do need to do this, you can simply use [`ActiveRecord::QueryMethods#none`](http://api.rubyonrails.org/classes/ActiveRecord/QueryMethods.html#method-i-none).

Your setup looks fine to me. I'm don't understand what the problem is.

**Comment by nruth** [comment]
> Two is that if I add a condition to check for an empty AR Relation and I pass in a class name to the policy_scoped method then it wines because the class doesn't have the empty? method defined...

This is an issue with active-record's API really. You're finding that a model mostly behaves like a scope, but not always. So pass in a scope instead: ClassName.all instead of ClassName?

**Comment by jnicklas** [comment]
Ahh, I think @nruth's comment is spot on. I'm closing this issue, since that seems like a good answer.

**Comment by erlandsona** [comment]
Thanks all for the replies... I'm not sure they necessarily solve the issue but I do think y'all are right about it being more of an active record issue...

The main motivation behind the problem was that sometimes the do was empty for a particular model and so I just needed to account for that behaviour and I was hoping Pundit might do that for me.

Primarily just because I didn't like what I had to write to make it work. But oh well such as life!

Thanks for the help!
